sdk/compute/Azure.ResourceManager.Compute/src/Customize/Models/GalleryDiskImage.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Compute.Models
+{
+    public partial class GalleryDiskImage
+    {
+        /// <summary>
+        /// The source for the disk image.
+        /// This is no longer supported. Please use <see cref="GallerySource"/> instead.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public GalleryArtifactVersionSource Source { get; set; } // the old property to keep backward compatiblility, working as a backing property of GallerySource
```

Comment:
Can we add unit tests around this. It appears that if someone does this it will fail
```c#
var galleryDiskImage = new GalleryDiskImage(...);
galleryDiskImage.Source = new GalleryArtifactVersionSource(...);
var gallerySource = galleryDiskImage.GallerySource; // <-- invalid cast exception?
```

Reply:
I added a test - but we will not get a cast exception, just get back null because we are using `as`